go/tasks/plugins/array/k8s/executor.go

```diff
@@ -91,12 +91,19 @@ func (e Executor) Handle(ctx context.Context, tCtx core.TaskExecutionContext) (c
 			return core.UnknownTransition, err
 		}
 
-		externalResources, err = arrayCore.InitializeExternalResources(ctx, tCtx, pluginState,
-			func(tCtx core.TaskExecutionContext, childIndex int) string {
-				subTaskExecutionID := NewSubTaskExecutionID(tCtx.TaskExecutionMetadata().GetTaskExecutionID(), childIndex, 0)
-				return subTaskExecutionID.GetGeneratedName()
-			},
-		)
+		// if the next state has not yet transitioned from PhaseStart it means that the background
+		// cache lookup for map task subtasks has not yet been completed. this is necessary for
+		// the InitializeExternalResources operation, which is used to notify admin of subtask
+		// including cache status. a transition from PhaseStart indicates cache lookup completion
+		// and we may InitializeExternalResources.
+		if np, _ := nextState.GetPhase(); np != arrayCore.PhaseStart {
```

Comment:
In go/tasks/plugins/array/awsbatch/executor.go, you checked explicitly for the transition from PhaseStart to "something that isn't phaseStart"... I"ll look around a bit more, but is "the previous state was PhaseStart" guaranteed  by virtue of execution reaching this line? Otherwise, is there a situation where this gets called, but the previous state wasn't Phase Start? Does that cause problems?

Reply:
Okay cool, the switch statement ensures the we were previously in PhaseStart, so the logic here replicates that in go/tasks/plugins/array/awsbatch/executor.go (i.e., ensuring a state transition from PhaseStart to something that isn't PhseStart)

Reply:
Correct.